pgoapi/pgoapi.py

```diff
@@ -226,23 +227,62 @@ def heartbeat(self):
         self._heartbeat_number += 1
         return res
 
-    def walk_to(self, loc, waypoints=[]):  # location in floats of course...
+    def _walk(self, loc, waypoints=[], directly=False):  # location in floats of course...
         steps = get_route(self._posf, loc, self.config.get("USE_GOOGLE", False), self.config.get("GMAPS_API_KEY", ""),
                           self.experimental and self.spin_all_forts, waypoints)
         catch_attempt = 0
+        base_travel_link = "https://www.google.com/maps/dir/%s,%s/" % (self._posf[0], self._posf[1])
+        step_size = self.config.get("STEP_SIZE", 200)
```

Comment:
the distance_in_meters gives the point to point distance. But if we use google maps then it may be higher since we may go around corners etc... How does this handle it?  because 
total_distance may be 1 km (point to point) but google (esp using waypoints) may give us 5km, and while we keep on adding distance to total_distance_travelled we may exceed the total_distance but still not end up where we wanted to be.